### PR TITLE
Support nvcc's `--threads` flag

### PR DESCRIPTION
**Pull request by trxcllnt**
This PR adds support for nvcc's [`--threads`](https://docs.nvidia.com/cuda/cuda-compiler-driver-nvcc/#threads-number-t) flag.

### CODE REVIEW

**Review by Xuanwo — APPROVED** [review]

**Review by drahnr — APPROVED** [review]

**Comment by sylvestre** [comment]
could you please run rustfmt on your change? thanks

**Comment by trxcllnt** [comment]
@sylvestre done, thanks.

About this PR, the `--threads` flag doesn't affect anything about the output (just how many device architectures `nvcc` compiles in parallel under the hood). Is there an elegant way to pass this flag through to `nvcc`, but not allow it to affect the computed hash?

This would prevent existing caches from being invalidated, but more importantly, it would allow devs (and CI machines) with different values for `--threads` to still share the same cache.

**Comment by trxcllnt** [comment]
I added a new `unhashed_args` vector + `Unhashed` arg type. Need to investigate how to test the new behavior besides running locally.

File: src/compiler/c.rs [hunk 1]
```diff
@@ -100,6 +100,8 @@ pub struct ParsedArguments {
     pub common_args: Vec<OsString>,
     /// Commandline arguments for the compiler that specify the architecture given
     pub arch_args: Vec<OsString>,
+    /// Commandline arguments for the preprocessor or the compiler that don't affect the computed hash.
+    pub unhashed_args: Vec<OsString>,
```
Comment (drahnr) [inline src/compiler/c.rs:104]: I'd like to have another statement, that none of these arguments listed may in any circumstance change the resulting binary.

**Review by drahnr — APPROVED** [review]
I think this is a good structural change.

Thank you!

**Pull request merged**